File: solutions.digamma.damas.jcr/src/test/java/solutions/digamma/damas/jcr/content/JcrFolderTest.java
package solutions.digamma.damas.jcr.content;

import org.junit.After;
import org.junit.Before;
import org.junit.Test;
import solutions.digamma.damas.auth.Token;
import solutions.digamma.damas.content.Folder;
import solutions.digamma.damas.content.FolderManager;
import solutions.digamma.damas.jcr.Mocks;
import solutions.digamma.damas.jcr.WeldTest;

import java.util.List;

/**
 * Test folder object.
 *
 * Created by Ahmad on 9/3/17.
 */
public class JcrFolderTest extends WeldTest {

    private Token token;
    private Folder folder;

    @Before
    public void setUp() throws Exception {
        FolderManager manager = inject(JcrFolderManager.class);
        this.token = this.login.login("admin", "admin");
        String parentId = manager
                .find(this.token).getObjects().iterator().next().getId();
        Folder parent = this.folder = manager.create(this.token,
                Mocks.folder(parentId, "test"));
        for (int i = 0; i < 10; i++) {
            parentId = parent.getId();
            String name = String.format("test_%d", i);
            parent = manager.create(this.token,
                    Mocks.folder(parentId, name));
        }
    }

    @After
    public void tearDown() throws Exception {
        this.login.logout(this.token);
    }

    @Test
    public void expand() throws Exception {
        this.folder.expand();

    }

    @Test
    public void expandContent() throws Exception {
        Folder expanded = this.folder;
        expanded.expandContent();
        for (int i = 0; i < 10; i++) {
            expanded = expanded.getContent().getFolders().iterator().next();
        }
    }

    @Test
    public void expandContentToDepth() throws Exception {
        Folder expanded = this.folder;
        int depth = 5;
        expanded.expandContent(depth);
        for (int i = 0; i < depth; i++) {
            expanded = expanded.getContent().getFolders().iterator().next();
        }
        assert expanded.getContent() == null;
    }

    @Test
    public void getContent() throws Exception {
        this.folder.expandContent();
        List<? extends Folder> folders = this.folder.getContent().getFolders();
        assert folders.size() != 0;
        Folder subfolder = folders.iterator().next();
        assert subfolder.getName().equals(String.format("test_%d", 0));
    }
}

File: solutions.digamma.damas.jcr/src/test/kotlin/solutions/digamma/damas/jcr/content/JcrFolderTest.kt
package solutions.digamma.damas.jcr.content

import org.junit.After
import org.junit.Before
import org.junit.Test
import solutions.digamma.damas.content.Folder
import solutions.digamma.damas.jcr.Mocks
import solutions.digamma.damas.jcr.RepositoryTest

/**
 * Test folder object.
 *
 * Created by Ahmad on 9/3/17.
 */
class JcrFolderTest : RepositoryTest() {

    private var folder: Folder? = null
    private var root: Folder? = null

    @Before
    @Throws(Exception::class)
    fun setUp() {
        val manager = inject(JcrFolderManager::class.java)
        this.login()
        this.root = manager.find("/")
        val parentId = this.root!!.id
        this.folder = manager.create(Mocks.folder(parentId, "test"))
        var parent = this.folder
        repeat(10) {
            parent = manager.create(Mocks.folder(parent!!.id, "test_$it"))
        }
    }

    @After
    @Throws(Exception::class)
    fun tearDown() {
        val manager = inject(JcrFolderManager::class.java)
        manager.delete(this.folder!!.id)
        this.logout()
    }

    @Test
    @Throws(Exception::class)
    fun expandContent() {
        var expanded = this.folder
        expanded!!.expandContent()
        repeat(10) {
            expanded = expanded!!.content.folders.iterator().next()
        }
    }

    @Test
    @Throws(Exception::class)
    fun expandContentToDepth() {
        var expanded = this.folder
        val depth = 5
        expanded!!.expandContent(depth)
        repeat(depth) {
            expanded = expanded!!.content.folders.iterator().next()
        }
        assert(expanded!!.content == null)
    }

    @Test
    @Throws(Exception::class)
    fun getContent() {
        this.folder!!.expandContent()
        val folders = this.folder!!.content.folders
        assert(folders.size != 0)
        val subfolder = folders.iterator().next()
        assert(subfolder.name == "test_${0}")
    }

    @Test
    fun pathIds() {
        var expanded = this.folder
        val pathIds = this.folder!!.pathIds.toMutableList()
        expanded!!.expandContent()
        repeat(10) {
            expanded = expanded!!.content.folders.iterator().next()
            pathIds.add(expanded!!.id)
        }
        assert(expanded!!.pathIds.equals(pathIds))
    }

    @Test
    fun pathIdsOnRoot() {
        assert(this.root!!.pathIds.size == 1)
        assert(this.root!!.pathIds[0].equals(this.root!!.id))
    }
}
